src/app/app.component.ts

```diff
@@ -59,15 +59,32 @@ export class MyApp {
     { title: 'Search', component: SearchPage }
   ]
 
+
   constructor(translate: TranslateService, platform: Platform, settings: Settings, config: Config) {
     // Set the default language for translation strings, and the current language.
     translate.setDefaultLang('en');
-    translate.use('en')
-
+    translate.use('en');
+    settings.getValue('language').then(val => {
+      if (!val) {
+        val = 'en';
+      }
+      translate.setDefaultLang(val);
+      translate.use(val);
+    });
     translate.get(['BACK_BUTTON_TEXT']).subscribe(values => {
       config.set('ios', 'backButtonText', values.BACK_BUTTON_TEXT);
     });
+    settings.settingsObservable.subscribe((value) => {
```

Comment:
Since this observable returns once when the settings are read, perhaps it should be renamed to `settings.load$`?